### ISSUE TITLE
Select mode: Display the block name on hover

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#31283]
Which block will be selected on click can sometimes feel like a mystery. In the Site Editor (and in Select Mode) we display outlines which help a bit, but is still inadequate in situations where a single block fills a container, and you need to click more precisely.

In such situations seeing the block name on hover can be a helpful indicator of precisely which block will be selected _before_ you click.

Here's an initial design concept:

![hover](https://user-images.githubusercontent.com/846565/116407191-6b2a6c80-a829-11eb-8b39-c498a6c4cc97.gif)

**Comment by karmatosed** [comment]
I think this does add a context but I am not sure it doesn't bring more readibility issues. For example there is now a word very close to another:

![image](https://user-images.githubusercontent.com/253067/116413006-1d186780-a82f-11eb-82e3-0695e9a3a346.png)

Maybe having to the side works when there is space? The styling also makes this look clickable like an actionable button.

**Comment by jameskoster** [comment]
If we place it to the side of the block we inevitably end up having to move it for narrow screens, or for blocks that fill the canvas horizontally. The different widths (based on the block name) create some visual awkwardness too:

![side](https://user-images.githubusercontent.com/846565/116416197-10494300-a832-11eb-8b9f-e20a21cbe8f8.gif)

**Comment by jameskoster** [comment]
> The styling also makes this look clickable like an actionable button.

It is :) You click the block to select it. Essentially the whole block is a button, and the label here is just an extension of that.

Edit: Here's a slightly less pixelly/noisy iteration that elaborates on the "entire block is a button" concept:

![block-as-button](https://user-images.githubusercontent.com/846565/116420053-8d29ec00-a835-11eb-800d-82c13ac0f434.gif)

**Comment by karmatosed** [comment]
Ah! It wasn't super clear in description and gif it was clickable, thanks for clarification. I think if you're going for the 'whole block is a button' you probably should go all in as with last iteration.

**Comment by paaljoachim** [comment]
Good idea, James!

**Comment by jasmussen** [comment]
We've tried it in the past, and removed it quickly as it was too noisy. There's something to the idea, but we should be very careful with the implementation. One idea is to make it a tooltip, identical to all other tooltips in the editor — that is, it appears after a brief delay, in context of the cursor, and is a black strip with white text.

**Comment by jameskoster** [comment]
Yes I think in Edit mode this would be too noisy. And agree that we need to exercise caution with any elements that obscure the canvas. It probably makes sense to let this one stew a bit and revisit as we explore other Select mode iterations.

**Comment by mtias** [comment]
Closing for now.
